server/common/dao/src/main/java/org/kaaproject/kaa/server/common/dao/service/CTLServiceImpl.java

```diff
@@ -554,6 +555,12 @@ private Schema parseDependencies(CTLSchemaDto schema, final Schema.Parser parser
                 Integer version = dependency.get(VERSION).getIntValue();
                 CTLSchemaDto child = this.findCTLSchemaByFqnAndVerAndTenantIdAndApplicationId(
                         fqn, version, parent.getMetaInfo().getTenantId(), parent.getMetaInfo().getApplicationId());
+                if (child == null) { // Not found within the application scope
+                    String tenantId = parent.getMetaInfo().getTenantId();
+                    String applicationId = null; // Search within the tenant scope then
+                    child = this.findAnyCTLSchemaByFqnAndVerAndTenantIdAndApplicationId(fqn, version, tenantId, applicationId);
```

Comment:
I believe that you can use thins method instead of "findCTLSchemaByFqnAndVerAndTenantIdAndApplicationId" and there will be no need to execute request twise (worst case scenario).